### PR TITLE
don't remove reactor_config plugin if config_map is provided

### PR DESCRIPTION
**Pull request by rcerven**
Signed-off-by: Robert Cerven <rcerven@redhat.com>

### CODE REVIEW

File: osbs/build/build_request.py [hunk 1]
```diff
@@ -228,7 +228,9 @@ def set_label(self, name, value):
         self.template['metadata']['labels'][name] = value
 
     def render_reactor_config(self):
-        if self.spec.reactor_config_secret.value is None:
+        if self.spec.reactor_config_secret.value is None and \
+                not self.spec.reactor_config_map.value and \
```
Comment (lcarva) [inline osbs/build/build_request.py]: It's a little confusing that for some values we're comparing to None while others verifying they're falsy. Can we make it the same? `is None` seems like a reasonable check for all.

**Comment by lcarva** [comment]
Needs unit tests.

File: osbs/build/build_request.py (hunk 1 above)
Comment (rcerven) [reply osbs/build/build_request.py]: ok

**Comment by rcerven** [comment]
unit test added

File: tests/build_/test_build_request.py [hunk 2]
```diff
@@ -3091,3 +3091,50 @@ def test_set_required_secrets(self, platforms, reactor_config_map,
...
+            'reactor_config_secret': reactor_config_secret,
+            'osbs_api': MockOSBSApi(),
+        }
+        build_request.set_params(**kwargs)
+        build_json = build_request.render()
+
+        plugins = get_plugins_from_build_json(build_json)
+
+        if reactor_config_secret is None and reactor_config_override is None and \
+                reactor_config_map is None:
```
Comment (twaugh) [inline tests/build_/test_build_request.py]: This is repeating the logic inside the test, but I don't see an easier way of writing it without introducing a truth table with all 8 possibilities.

**Comment by twaugh** [comment]
Typo in commit message ("conig")

**Comment by rcerven** [comment]
fixed typo in commit msg

**Comment by lcarva** [comment]
Not updating release notes draft because this is bugfix to an unreleased feature.

**Pull request merged**